src/sync/contents/rustdoc/code_block.rs

```diff
@@ -109,4 +132,54 @@ mod tests {
         check("edition99999", "edition99999", false);
         check("editionabcd", "editionabcd", false);
     }
+
+    #[test]
+    fn hide_codeblock_line() {
+        let input = r#"
+Lorem ipsum
+
+```rust
+# This line and the next should be hidden, but the following should not.
+#
+#[derive(Debug)]
+struct Foo;
+
+fn main() {
+    # As should this and the next line.
+    #
+    #But not this.
+    ## This should become a single #.
+    ##And this.
+}
+```
+
+```toml
+# This is not Rust so it should not be hidden.
+```
+"#;
+
+        let expected = r#"Lorem ipsum
+
+````rust
+#[derive(Debug)]
+struct Foo;
+
+fn main() {
+    #But not this.
+    # This should become a single #.
+    #And this.
+}
+````
+
+````toml
+# This is not Rust so it should not be hidden.
+````"#;
+
+        let events: Vec<_> = pulldown_cmark::Parser::new(input).collect();
+        let events: Vec<_> = super::convert(events).collect();
+
+        let mut output = String::new();
+        pulldown_cmark_to_cmark::cmark(events.into_iter(), &mut output).unwrap();
+        assert_eq!(output, expected, "output matches expected");
+    }
```

Comment:
I couldn't find any fixture-based/data-driven tests in the repo, so I did this ad-hoc thing.